text/0000-reseeding-rand.md

```diff
@@ -0,0 +1,682 @@
+- Start Date: 2013-01-23
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Stabilise the `std::rand` module by focusing it on a smaller set of
+functionality, moving "fancy" functionality to a crates.io crate.
+
+# Motivation
+
+The `std::rand` module provides various facilities for working with
+random numbers and randomness in general. This is a surprisingly
+sensitive topic, with different use cases. `std` does not aim to
+completely satisfy everything, but rather aims to provide a minimal
+set of functionality that tries to be simple and straightforward
+without restricting people building on it, and this reasoning extends
+to `std::rand`.
+
+A major area of use for randomness is cryptography where the
+recommended approach is to use the operating system if at all
+possible. There is no particular reason for `std::rand` to deviate
+from this. However, other uses of randomness (e.g. simulations) do not
+need the cryptographical strength, and prefer the much improved
+performance one can get from using a weaker RNG. Similarly, a use-case
+like seeding a hashmap to avoid algorithmic complexity DoS attacks
+does not require a (possibly expensive) call into the operating system
+for every `HashMap::new()` call; a high-quality user-space RNG with an
+unpredictable seed (e.g. from the OS) is almost certainly good enough.
+
+To justify the inclusion of non-OS interfaces, the following benchmark
+was run on my Linux 3.16 system (before the `getrandom(2)` syscall was
```

Comment:
The `getrandom` call isn't about performance. It uses the same pool and requires the same number of system calls. It exists because it works in an environment like a sandbox without random devices and because it was able to implement the age old missing feature of blocking until the initial seeding is complete.